site/app/libraries/database/DatabaseQueries.php

```diff
@@ -2561,7 +2561,7 @@ public function getUsersByRotatingSections($sections, $orderBy = "rotating_secti
      * @return array
      */
     public function getRegistrationSections() {
-        $this->course_db->query("SELECT * FROM sections_registration ORDER BY SUBSTRING(sections_registration_id, '^[^0-9]*'), COALESCE(SUBSTRING(sections_registration_id, '[0-9]+')::INT, -1), SUBSTRING(sections_registration_id, '[^0-9]*$') ");
+        $this->course_db->query("SELECT * FROM sections_registration ORDER BY SUBSTRING(sections_registration_id, '^[^0-9]*'), CAST(SUBSTRING(sections_registration_id, '[0-9]+') AS NUMERIC), SUBSTRING(sections_registration_id, '[^0-9]*$')");
```

Comment:
What is the reason to change from COALESCE to CAST, I do see that the functionality is same, but there is also other functions that use identical code, would it be better to switch other queries as well?

Reply:
In this query, CAST is being used to convert a substring extracted from "sections_registration_id" into a numeric value because I had to deal with the case that if 20 digit long number in entered, INT or Bigint are not able to handle it.

Reply:
That is true, ::INT will convert to PHP int max, which is 2147483647, I tested it with our current query, and it does fail after 2147483647. However, with your query (using CAST) I am getting below error. 
![Screen Shot 2023-08-05 at 10 45 02 AM](https://github.com/Submitty/Submitty/assets/123261952/826c8d36-24ff-4220-b0fc-b91089a6ab73)